libraries/AC_Avoidance/AP_OADijkstra.cpp

```diff
@@ -133,27 +148,39 @@ AP_OADijkstra::AP_OADijkstra_State AP_OADijkstra::update(const Location &current
         }
     }
 
-    // rebuild path if destination has changed
-    if (!destination.same_latlon_as(_destination_prev)) {
+    // rebuild path if destination or next_destination has changed
+    if (!destination.same_latlon_as(_destination_prev) || !next_destination.same_latlon_as(_next_destination_prev)) {
         _destination_prev = destination;
+        _next_destination_prev = next_destination;
         _shortest_path_ok = false;
     }
 
     // calculate shortest path from current_loc to destination
     if (!_shortest_path_ok) {
         _shortest_path_ok = calc_shortest_path(current_loc, destination, error_id);
         if (!_shortest_path_ok) {
+            dest_to_next_dest_clear = _dest_to_next_dest_clear = false;
             report_error(error_id);
             Write_OADijkstra(DIJKSTRA_STATE_ERROR, (uint8_t)error_id, 0, 0, destination, destination);
             return DIJKSTRA_STATE_ERROR;
         }
         // start from 2nd point on path (first is the original origin)
         _path_idx_returned = 1;
+
+        // check if path from destination to next_destination intersects with a fence
+        _dest_to_next_dest_clear = false;
+        if (!next_destination.is_zero()) {
+            Vector2f seg_start, seg_end;
+            if (destination.get_vector_xy_from_origin_NE(seg_start) && next_destination.get_vector_xy_from_origin_NE(seg_end)) {
+                _dest_to_next_dest_clear = !intersects_fence(seg_start, seg_end);
+            }
+        }
     }
 
     // path has been created, return latest point
     Vector2f dest_pos;
-    if (get_shortest_path_point(_path_idx_returned, dest_pos)) {
+    uint8_t path_length = get_shortest_path_numpoints() - 1;
```

Comment:
could get_shortest_path_numpoints() be zero so this wraps to 255?

Reply:
Thanks for noticing this.  I've added protection against the wrap even though it is not currently possible to get to this point in the code with a zero length path.